### PR TITLE
Latest layer 3 Oasys assessment

### PR DESCRIPTION
**Pull request by stevomcallister**

### CODE REVIEW

File: projects/approved-premises-and-oasys/deploy/values-dev.yml [hunk 1]
```diff
@@ -1,10 +1,8 @@
 environment_name: delius-test
 
-enabled: false # TODO set this to true when you're ready to deploy your service
-
 env:
   SENTRY_ENVIRONMENT: dev
-  SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_HMPPS-AUTH_TOKEN-URI: #TODO complete with OASYS client token uri
+  SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_ORDS-OASYS-AUTH_TOKEN-URI: https://ords.t1.oasys.service.justice.gov.uk/eor/oasys/oauth/token
```
Comment (marcus-bcl) [inline projects/approved-premises-and-oasys/deploy/values-dev.yml:5]: Did we ever decide on T1 vs T2...?

**Review by marcus-bcl — DISMISSED** [review]
Looks good!

File: .idea/runConfigurations/approved_premises_and_oasys.xml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="approved-premises-and-oasys" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot">
     <option name="ACTIVE_PROFILES" value="dev" />
+    <envs>
+      <env name="integrations.ords-oasys.url" value="https://ords.t1.oasys.service.justice.gov.uk/eor/oasys" />
+    </envs>
```
[the remark this answers is not in the document]
Comment (stevomcallister) [reply .idea/runConfigurations/approved_premises_and_oasys.xml:6]: oops - that's a mistake

File: projects/approved-premises-and-oasys/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/oasys/OasysClient.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.cloud.openfeign.FeignClient
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import uk.gov.justice.digital.hmpps.config.FeignOAuth2Config
+
+@FeignClient(
+    name = "ords-oasys",
+    url = "\${integrations.ords-oasys.url}",
+    configuration = [FeignOAuth2Config::class],
+    decode404 = true
```
[the remark this answers is not in the document]
Comment (stevomcallister) [reply projects/approved-premises-and-oasys/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/oasys/OasysClient.kt]: yes it does - have put in a change for this

File: projects/approved-premises-and-oasys/src/main/kotlin/uk/gov/justice/digital/hmpps/controller/AssessmentController.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.integrations.oasys.OasysAssessment
+
+@RestController
+class AssessmentController(private var oasysAssessmentService: OasysAssessmentService) {
+    @PreAuthorize("hasRole('ROLE_PROBATION')")
```
[the remark this answers is not in the document]
Comment (stevomcallister) [reply projects/approved-premises-and-oasys/src/main/kotlin/uk/gov/justice/digital/hmpps/controller/AssessmentController.kt:11]: ok

File: projects/approved-premises-and-oasys/deploy/values-dev.yml (hunk 1 above)
Comment (stevomcallister) [reply projects/approved-premises-and-oasys/deploy/values-dev.yml:6]: updated now

**Pull request merged**